File: python/Move Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        zeroCount = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                zeroCount += 1
            else:
                nums[i - zeroCount] = nums[i]
        for i in range(1, zeroCount + 1):
            nums[-i] = 0

File: cpp/Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 0) {
                for (int j = i + 1; j < nums.size(); j++) {
                    if (nums[j] != 0) {
                        nums[i] = nums[j];
                        nums[j] = 0;
                        break;
                    }
                }
            }
        }
    }
};
